src/literacy.rs

```diff
@@ -0,0 +1,123 @@
+#[cfg(all(feature = "std", feature = "use-core2"))]
+compile_error!("feature \"std\" and \"use-core2\" cannot be enabled together.");
```

Comment:
I think we should "default" to core2 in this case - if we already are building against the core2 dep, we might as well just assume the user set core2/std and then the core2 impl is the same as doing the std impl.

Reply:
I agree but there are some implementation issue, in particular

> we might as well just assume the user set core2/std

How so? with another feature `use-core2-std = ["core2/std"]`? This would cause other issues in feature combinations...
The other option, using `std = ["core2/std"]` will cause problem in default features compilation on rust 1.29

Reply:
> How so? with another feature use-core2-std = ["core2/std"]? This would cause other issues in feature combinations...

We don't have to specify it ourselves for core2/std to be enabled by some other dependency path. Its probably neater to just impl for std and not core2 than to compile-error, and if users intended to use core2 they can always fix it by, themselves, directly requiring core2/std. This avoids the issue of having some crate that depends on A and B and A enables std and B enables core2, giving the user no way to fix the issue aside from patching either A or B.

Reply:
Sure, removed the `compile_error` and implementing std impl only if there is std but not core2